### PR TITLE
Refactored the mailers

### PR DESCRIPTION
**Pull request by KevinMulhern**
Fixes: https://github.com/TheOdinProject/theodinproject/issues/323 and https://github.com/TheOdinProject/theodinproject/issues/322

### CODE REVIEW

File: app/mailers/contact_mailer.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-
-  def suggestion_email(suggestion_text, url="", user_email="")
+  def suggestion_email(suggestion_text, url='', user_email='')
     @url = url
     @suggestion_text = suggestion_text
     @user_email = user_email
-    mail( { :to => FEEDBACK_EMAIL,
-            :subject => "Site FEEDBACK" } )
+
+    mail(:to => feedback_email, :subject => 'Site FEEDBACK')
```
Comment (arun1595) [inline app/mailers/contact_mailer.rb]: Can we use ruby 1.9 hash syntax please?

Comment (KevinMulhern) [inline app/mailers/contact_mailer.rb]: Nice catch, on it 👍

File: app/mailers/contact_mailer.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
 class ContactMailer < ActionMailer::Base
-  default from: "The Odin Project <admin@theodinproject.com>"
-  FEEDBACK_EMAIL = "feedback@theodinproject.com"
-  CONTACT_EMAIL = "contact@theodinproject.com"
-  ODIN_URL = "http://www.theodinproject.com"
+  default from: 'The Odin Project <admin@theodinproject.com>'
 
-
-  def suggestion_email(suggestion_text, url="", user_email="")
+  def suggestion_email(suggestion_text, url='', user_email='')
```
Comment (arun1595) [inline app/mailers/contact_mailer.rb]: Why do we need default values here?

Comment (KevinMulhern) [inline app/mailers/contact_mailer.rb]: Its possible for a non logged in user to send an suggestion email, The url one I'm not so sure about it. It's the url of the page the user sends the suggestion from. So there should theoretically always be a url.

Comment (arun1595) [inline app/mailers/contact_mailer.rb]: I just had a look at `StaticPagesController`. Looks like we'll always always have a `url` and `user_email`(not always an email) too. Please checkout `suggestion` and `user_identifier` methods in `static_pages_controller.rb`

Comment (KevinMulhern) [inline app/mailers/contact_mailer.rb]: Yup you're right, on it 👍

Comment (arun1595) [inline app/mailers/contact_mailer.rb]: I think we don't really need default values here. It'll better to change the `user_identifier` method as

```ruby
def user_identifier
  current_user_email || 'Anonymous'
end
```

What do you think?

File: spec/mailers/contact_mailer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe ContactMailer, :type => :mailer do
```
Comment (arun1595) [inline spec/mailers/contact_mailer_spec.rb]: Nit. Ruby 1.9 hash syntax please.

File: spec/mailers/user_mailer_spec.rb [hunk 4]
```diff
@@ -1,5 +1,53 @@
 require "rails_helper"
 
-describe UserMailer do
+RSpec.describe UserMailer, :type => :mailer do
```
Comment (arun1595) [inline spec/mailers/user_mailer_spec.rb]: Same here.

**Pull request merged**